worksheet_D/TicTacToe.py

```diff
@@ -0,0 +1,114 @@
+class OxoBoard:
+
+    def __init__(self,x,y):
+        self.x = x
+        self.y = y
+        self.board = {(0,0):0,(0,1):0,(0,2):0,(1,0):0,(2,0):0,(1,1):0,(2,2):0,(1,2):0,(2,1):0}
+
+    def get_square(self, x, y):
+        return self.board[x,y]
+
+    def set_square(self, x, y, full):
+        if self.board[x,y] == 0:
+            self.board[x,y] = full
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        spaces = 9
+        for i in xrange(0,3):
+            for j in xrange(0,3):
+                if self.board[i, j] > 0:

```

Comment:
OK, but not the most efficient approach -- you can return `False` as soon as you see a square containing a 0, but here you have to scan to the end of the board.